File: javascript/3.js
function longestCommonPrefix(strs) {
   if (!strs.length)
       return "";

   let prefix = strs[0];
   for (let i = 1; i < strs.length; i++) {
       while (strs[i].indexOf(prefix) !== 0) {
           prefix = prefix.substring(0, prefix.length - 1);
       }
   }

   return prefix;
}

let strs = ["flower", "flow", "flight"];
console.log("\n\n\n\nCorrect " + longestCommonPrefix(strs) + "\n\n");
 

File: rust/3.rs
fn longest_common_prefix(strs: Vec<&str>) -> String {
   if strs.is_empty() {
       return String::new();
   }

   let mut prefix = strs[0].to_string();
   for s in &strs[1..] {
       while !s.starts_with(&prefix) {
           prefix.pop();
       }
   }

   prefix
}

fn main() {
   let strs = vec!["flower", "flow", "flight"];
   println!("\n\n\n\nCorrect {}\n\n", longest_common_prefix(strs));
}
 